src/hotspot/cpu/x86/x86.ad

```diff
@@ -9233,6 +9233,20 @@ instruct vfma_mem_masked(vec dst, vec src2, memory src3, kReg mask) %{
   ins_pipe( pipe_slow );
 %}
 
+instruct evcompress_reg(vec dst, vec src, kReg mask) %{
+  predicate(UseAVX > 2);
+  match(Set dst (CompressV src mask));
+  format %{ "vcompress $dst, $src, $mask" %}
+  ins_encode %{
+    assert(Matcher::vector_length_in_bytes(this) >= 16, "sanity");
+    int vector_len = vector_length_encoding(this);
+    BasicType etype = Matcher::vector_element_basic_type(this);
+    // Zeroing-masking
+    __ evcompress(etype, $dst$$XMMRegister, $mask$$KRegister, $src$$XMMRegister, false /*merge*/, vector_len);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
```

Comment:
Did you intend this to be part of pull 144.

Reply:
Done